---
id: v3-ht-annotations
title: 'gnomAD v4 genomes Hail Table annotation descriptions'
---

The gnomAD v3 Hail Table annotations are defined below:

**Global fields**:

- **freq_meta**: Allele frequency metadata. An ordered list containing the frequency aggregation group for each element of the ‘freq’ array row annotation.
- **freq_index_dict**: Dictionary keyed by specified label grouping combinations (group: adj/raw, pop: gnomAD inferred global population, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the ‘freq’ array row annotation.
- **faf_index_dict**: Dictionary keyed by specified label grouping combinations (group: adj/raw, pop: gnomAD inferred global population, sex: sex karyotype), with values describing the corresponding index of each grouping entry in the filtering allele frequency (‘faf’) row annotation.
- **faf_meta**: Filtering allele frequency metadata. An ordered list containing the frequency aggregation group for each element of the ‘faf’ array row annotation.
- **vep_globals**: Information about VEP annotations. 
  - **vep_version**: VEP version that was run on the callset.
  - **vep_csq_header**: VEP header for VCF export.
  - **vep_help**: Output from vep --help.
  - **vep_config**: VEP configuration to run VEP with [Hail](https://hail.is/docs/0.2/methods/genetics.html#hail.methods.vep).
- **filtering_model**: The variant filtering model used and its specific cutoffs.
  - **model_name**: Variant filtering model name used in the 'filters' row annotation, indicating the variant was filtered by this model during variant QC.
  - **score_name**: Annotation name of the score used for variant filtering.
  - **snv_cutoff**: SNV filtering cutoff information.
    - **bin**: Filtering percentile cutoff for SNVs.
    - **min_score**: Minimum score at SNV filtering percentile cutoff.
  - **indel_cutoff**: Indel filtering cutoff information.
    - **bin**: Filtering percentile cutoff for indels.
    - **min_score**: Minimum score at indel filtering percentile cutoff.
  - **model_id**: Variant filtering model ID for score data (used for internal specification of the model).
  - **snv_training_variables**: Variant annotations used as features in the SNV filtering model.
  - **indel_training_variables**: Variant annotations used as features in the indel filtering model.
- **age_distribution**: Callset-wide age histogram calculated on release samples.
  - **bin_edges**: Bin edges for the age histogram.
  - **bin_freq**: Bin frequencies for the age histogram. This is the number of records found in each bin.
  - **n_smaller**: Count of age values falling below lowest histogram bin edge.
  - **n_larger**: Count of age values falling above highest histogram bin edge.
- **freq_meta_sample_count**: A sample count per sample grouping defined in the 'freq_meta' global annotation.
- **vrs_version**: The Variant Representation Specification that was used to compute IDs on the callset.
  - **vrs-python_version**: The [version](https://github.com/ga4gh/vrs-python/tags) of the vrs-python library that was used to compute IDs on the callset.
  - **seqrepo_version**: The [version](https://github.com/biocommons/biocommons.seqrepo) of the SeqRepo database that was used in VRS computations.
- **tool_versions**: Versions of in silico predictors used in the callset.
  - **dbsnp_version**: dbSNP version used in the callset.
  - **revel_version**: Rare Exome Variant Ensemble Learner (REVEL) version.
  - **primateai_3d_version**: PrimateAI-3D version.
  - **cadd_version**: Combined Annotation Dependent Depletion (CADD) version.
  - **sift_version**: Sorting Intolerant From Tolerant (SIFT) version.
  - **polyphen_version**: Polymorphism Phenotyping v2 (Polyphen-v2) version.
  - **phylop_version**: PhyloP version.
  - **splice_ai_version**: SpliceAI version.
  - **pangolin_version**: Pangolin version.
  - **README**: Dictionary explaining the `freq_index_dict` and `freq_meta` global annotations.
  
**Row fields**:

- **locus**: Variant locus. Contains contig and position information.
- **alleles**: Variant alleles.
- **freq**: Array of allele frequency information (AC, AN, AF, homozygote count) for each frequency aggregation group in the gnomAD release.
  - **AC**: Alternate allele count in release.
  - **AF**: Alternate allele frequency, (AC/AN), in release.
  - **AN**: Total number of alleles in release.
  - **homozygote_count**: Count of homozygous alternate individuals in release.
- **raw_qual_hists**: Genotype quality metric histograms for all genotypes as opposed to high quality genotypes.
  - **gq_hist_all**: Histogram for GQ calculated on all genotypes.
    - **bin_edges**: Bin edges for the GQ histogram calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
    - **bin_freq**: Bin frequencies for the GQ histogram calculated on all genotypes. The number of records found in each bin.
    - **n_smaller**: Count of GQ values falling below lowest histogram bin edge, for GQ calculated on all genotypes.
    - **n_larger**: Count of GQ values falling above highest histogram bin edge, for GQ calculated on all genotypes.
  - **dp_hist_all**: Histogram for DP calculated on all genotypes.
    - **bin_edges**: Bin edges for the DP histogram calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100
    - **bin_freq**: Bin frequencies for the DP histogram calculated on all genotypes. The number of records found in each bin.
    - **n_smaller**: Count of DP values falling below lowest histogram bin edge, for DP calculated on all genotypes.
    - **n_larger**: Count of DP values falling above highest histogram bin edge, for DP calculated on all genotypes.
  - **gq_hist_alt**: Histogram for GQ in heterozygous individuals calculated on all genotypes.
    - **bin_edges**: Bin edges for the histogram of GQ in heterozygous individuals calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
    - **bin_freq**: Bin frequencies for the histogram of GQ in heterozygous individuals calculated on all genotypes. The number of records found in each bin.
    - **n_smaller**: Count of GQ values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
    - **n_larger**: Count of GQ values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.
  - **dp_hist_alt**: Histogram for DP in heterozygous individuals calculated on all genotypes.
    - **bin_edges**: Bin edges for the histogram of DP in heterozygous individuals calculated on all genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
    - **bin_freq**: Bin frequencies for the histogram of DP in heterozygous individuals calculated on all genotypes. The number of records found in each bin.
    - **n_smaller**: Count of DP values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
    - **n_larger**: Count of DP values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.
  - **ab_hist_alt**: Histogram for AB in heterozygous individuals calculated on all genotypes.
    - **bin_edges**: Bin edges for the histogram of AB in heterozygous individuals calculated on all genotypes are: 0.00|0.05|0.10|0.15|0.20|0.25|0.30|0.35|0.40|0.45|0.50|0.55|0.60|0.65|0.70|0.75|0.80|0.85|0.90|0.95|1.00.
    - **bin_freq**: Bin frequencies for the histogram of AB in heterozygous individuals calculated on all genotypes. The number of records found in each bin.
    - **n_smaller**: Count of AB values in heterozygous individuals falling below lowest histogram bin edge, calculated on all genotypes.
    - **n_larger**: Count of AB values in heterozygous individuals falling above highest histogram bin edge, calculated on all genotypes.
- **popmax**: Allele frequency information (AC, AN, AF, homozygote count) for the non-bottlenecked population with maximum allele frequency. Excludes Amish (ami), Ashkenazi Jewish (asj), European Finnish (fin), Middle Eastern (mid), and "Other" (oth) populations.
  - **AC**: Alternate allele count in the population with the maximum allele frequency.
  - **AF**: Maximum alternate allele frequency, (AC/AN), across populations in gnomAD.
  - **AN**: Total number of alleles in the population with the maximum allele frequency.
  - **homozygote_count**: Count of homozygous individuals in the population with the maximum allele frequency.
  - **pop**: Population with maximum allele frequency
  - **faf95**: Filtering allele frequency (using Poisson 95% CI) for the population with the maximum allele frequency.
- **qual_hists**: Genotype quality metric histograms for high quality genotypes.
  - **gq_hist_all**: Histogram for GQ calculated on high quality genotypes.
    - **bin_edges**: Bin edges for the GQ histogram calculated on high quality genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
    - **bin_freq**: Bin frequencies for the GQ histogram calculated on high quality genotypes. The number of records found in each bin.
    - **n_smaller**: Count of GQ values falling below the lowest histogram bin edge, calculated on high quality genotypes.
    - **n_larger**: Count of GQ values falling above the highest histogram bin edge, calculated on high quality genotypes.
  - **dp_hist_all**: Histogram for DP calculated on high quality genotypes.
    - **bin_edges**: Bin edges for the DP histogram calculated on high quality genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
    - **bin_freq**: Bin frequencies for the DP histogram calculated on high quality genotypes. The number of records found in each bin.
    - **n_smaller**: Count of DP values falling below the lowest histogram bin edge, calculated on high quality genotypes.
    - **n_larger**: Count of DP values falling above the highest histogram bin edge, calculated on high quality genotypes.
  - **gq_hist_alt**: Histogram for GQ in heterozygous individuals calculated on high quality genotypes.
    - **bin_edges**: Bin edges for the histogram of GQ in heterozygous individuals calculated on high quality genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
    - **bin_freq**: Bin frequencies for the histogram of GQ in heterozygous individuals calculated on high quality genotypes. The number of records found in each bin.
    - **n_smaller**: Count of GQ values in heterozygous individuals falling below the lowest histogram bin edge, calculated on high quality genotypes.
    - **n_larger**: Count of GQ values in heterozygous individuals falling above the highest histogram bin edge, calculated on high quality genotypes.
  - **dp_hist_alt**: Histogram for DP in heterozygous individuals calculated on high quality genotypes.
    - **bin_edges**: Bin edges for the histogram of DP in heterozygous individuals calculated on high quality genotypes are: 0|5|10|15|20|25|30|35|40|45|50|55|60|65|70|75|80|85|90|95|100.
    - **bin_freq**: Bin frequencies for the histogram of DP in heterozygous individuals calculated on high quality genotypes. The number of records found in each bin.
    - **n_smaller**: Count of DP values in heterozygous individuals falling below the lowest histogram bin edge, calculated on high quality genotypes.
    - **n_larger**: Count of DP values in heterozygous individuals falling above highest histogram bin edge, calculated on high quality genotypes.
  - **ab_hist_alt**: Histogram for AB in heterozygous individuals calculated on high quality genotypes.
    - **bin_edges**: Bin edges for the histogram of AB in heterozygous individuals calculated on high quality genotypes are: 0.00|0.05|0.10|0.15|0.20|0.25|0.30|0.35|0.40|0.45|0.50|0.55|0.60|0.65|0.70|0.75|0.80|0.85|0.90|0.95|1.00.
    - **bin_freq**: Bin frequencies for the histogram of AB in heterozygous individuals calculated on high quality genotypes. The number of records found in each bin.
    - **n_smaller**: Count of AB values in heterozygous individuals falling below the lowest histogram bin edge, calculated on high quality genotypes.
    - **n_larger**: Count of AB values in heterozygous individuals falling above the highest histogram bin edge, calculated on high quality genotypes.
- **faf**: Filtering allele frequency.
  - **faf95**: Filtering allele frequency (using Poisson 95% CI).
  - **faf99**: Filtering allele frequency (using Poisson 99% CI).
- **a_index**: The original index of this alternate allele in the multiallelic representation (1 is the first alternate allele or the only alternate allele in a biallelic variant).
- **was_split**: True if this variant was originally multiallelic, otherwise False.
- **rsid**: dbSNP reference SNP identification (rsID) numbers.
- **filters**: Variant filters; AC0: Allele count is zero after filtering out low-confidence genotypes (GQ &lt; 20; DP &lt; 10; and AB &lt; 0.2 for het calls), AS_VQSR: Failed allele-specific VQSR filtering thresholds of -2.7739 for SNPs and -1.0606 for indels, InbreedingCoeff: GATK InbreedingCoeff &lt; -0.3, PASS: Passed all variant filters.
- **info**: Struct containing typical GATK allele-specific (AS) info fields and additional variant QC fields.
  - **QUALapprox**: Sum of PL[0] values; used to approximate the QUAL score.
  - **SB**: Per-sample component statistics which comprise the Fisher's exact test to detect strand bias. Values are: depth of reference allele on forward strand, depth of reference allele on reverse strand, depth of alternate allele on forward strand, depth of alternate allele on reverse strand.
  - **MQ**: Root mean square of the mapping quality of reads across all samples.
  - **MQRankSum**: Z-score from Wilcoxon rank sum test of alternate vs. reference read mapping qualities.
  - **VarDP**: Depth over variant genotypes (does not include depth of reference samples).
  - **AS_ReadPosRankSum**: Allele-specific z-score from Wilcoxon rank sum test of alternate vs. reference read position bias.
  - **AS_pab_max**: Maximum p-value over callset for binomial test of observed allele balance for a heterozygous genotype, given expectation of 0.5.
  - **AS_QD**: Allele-specific variant call confidence normalized by depth of sample reads supporting a variant.
  - **AS_MQ**: Allele-specific root mean square of the mapping quality of reads across all samples.
  - **QD**: Variant call confidence normalized by depth of sample reads supporting a variant.
  - **AS_MQRankSum**: Allele-specific z-score from Wilcoxon rank sum test of alternate vs. reference read mapping qualities.
  - **FS**: Phred-scaled p-value of Fisher's exact test for strand bias.
  - **AS_FS**: Allele-specific phred-scaled p-value of Fisher's exact test for strand bias.
  - **ReadPosRankSum**: Z-score from Wilcoxon rank sum test of alternate vs. reference read position bias.
  - **AS_QUALapprox**: Allele-specific sum of PL[0] values; used to approximate the QUAL score.
  - **AS_SB_TABLE**: Allele-specific forward/reverse read counts for strand bias tests.
  - **AS_VarDP**: Allele-specific depth over variant genotypes (does not include depth of reference samples).
  - **AS_SOR**: Allele-specific strand bias estimated by the symmetric odds ratio test.
  - **SOR**: Strand bias estimated by the symmetric odds ratio test.
  - **singleton**: Variant is seen once in the callset.
  - **transmitted_singleton**: Variant was a callset-wide doubleton that was transmitted within a family from a parent to a child (i.e., a singleton amongst unrelated samples in cohort).
  - **omni**: Variant is present on the Omni 2.5 genotyping array and found in 1000 Genomes data.
  - **mills**: Indel is present in the Mills and Devine data.
  - **monoallelic**: All samples are homozygous alternate for the variant.
  - **AS_VQSLOD**: Allele-specific log-odds ratio of being a true variant versus being a false positive under the trained VQSR Gaussian mixture model.
  - **InbreedingCoeff**: Inbreeding coefficient, the excess heterozygosity at a variant site, computed as 1 - (the number of heterozygous genotypes) / (the number of heterozygous genotypes expected under Hardy-Weinberg equilibrium).
  - **vrs**: Struct containing information related to the Global Alliance for Genomic Health (GA4GH) Variant Representation Specification ([VRS](https://vrs.ga4gh.org/en/stable/)) standard.
    - **VRS_Allele_IDS**: The computed identifiers for the GA4GH VRS Alleles corresponding to the values in the alleles column.
    - **VRS_Starts**: Interresidue coordinates used as the location starts for the GA4GH VRS Alleles corresponding to the values in the alleles column.
    - **VRS_Ends**: Interresidue coordinates used as the location ends for the GA4GH VRS Alleles corresponding to the values in the alleles column.
    - **VRS_States**: The literal sequence states used for the GA4GH VRS Alleles corresponding to the values in the alleles column.
- **vep**: Consequence annotations from Ensembl VEP. More details about VEP output is described [here](https://ensembl.org/info/docs/tools/vep/vep_formats.html#output). VEP was run using the LOFTEE plugin and information about the additional LOFTEE annotations can be found [here](https://github.com/konradjk/loftee).
- **vqsr**: VQSR related variant annotations.
  - **AS_VQSLOD**: Allele-specific log-odds ratio of being a true variant versus being a false positive under the trained VQSR Gaussian mixture model.
  - **AS_culprit**: Allele-specific worst-performing annotation in the VQSR Gaussian mixture model.
  - **NEGATIVE_TRAIN_SITE**: Variant was used to build the negative training set of low-quality variants for VQSR.
  - **POSITIVE_TRAIN_SITE**: Variant was used to build the positive training set of high-quality variants for VQSR.
- **region_flag**: Struct containing flags for problematic regions.
  - **lcr**: Variant falls within a low complexity region.
  - **segdup**: Variant falls within a segmental duplication region.
- **allele_info**: Allele information.
  - **variant_type**: Variant type (snv, indel, multi-snv, multi-indel, or mixed).
  - **allele_type**: Allele type (snv, insertion, deletion, or mixed).
  - **n_alt_alleles**: Total number of alternate alleles observed at variant locus.
  - **was_mixed**: Variant type was mixed.
- **age_hist_het**: Histogram for age in all heterozygous release samples calculated on high quality genotypes.
  - **bin_edges**: Bin edges for the age histogram.
  - **bin_freq**: Bin frequencies for the age histogram. This is the number of records found in each bin.
  - **n_smaller**: Count of age values falling below lowest histogram bin edge.
  - **n_larger**: Count of age values falling above highest histogram bin edge.
- **age_hist_hom**: Histogram for age in all homozygous release samples calculated on high quality genotypes.
  - **bin_edges**: Bin edges for the age histogram.
  - **bin_freq**: Bin frequencies for the age histogram. This is the number of records found in each bin.
  - **n_smaller**: Count of age values falling below lowest histogram bin edge.
  - **n_larger**: Count of age values falling above highest histogram bin edge.
- **in_silico_predictors**: Variant prediction annotations. Struct contains prediction scores from multiple in silico predictors for variants that are predicted to be missense, impacting protein function, evolutionarily conserved, or splice-altering. We chose scores for either MANE Select or canonical transcripts if a prediction score was available for multiple transcripts.
  - **revel**: Rare Exome Variant Ensemble Learner (REVEL) [score](https://www.cell.com/ajhg/fulltext/S0002-9297(16)30370-6). An ensemble score for predicting the pathogenicity of missense variants (based on 13 other variant predictors).
    - **revel_max**: Score predicting the pathogenicity of missense variants with a range of 0 to 1. Variants with higher scores are predicted to be more likely to be deleterious. We prioritize max scores for MANE Select transcripts where possible and otherwise report a score for the canonical transcript.
  - **primateai_3d**: A deep neural network-trained [score](https://www.science.org/doi/10.1126/science.abn8197) for classifying pathogenicity of missense variants.
    - **primateai-3d_max**: Score ranging from 0 to 1 predicting the deleteriousness of missense variants. Higher scores indicate a variant is more likely to be deleterious. We prioritize max scores for MANE Select transcripts where possible and otherwise report a score for the canonical transcript. 
  - **cadd**: Combined Annotation Dependent Deletion [score](https://academic.oup.com/nar/article/47/D1/D886/5146191) used to predict deleteriousness of SNVs and indels.
    - **phred**: CADD Phred-like scaled C-scores ranging from 1 to 99 based on the rank of each variant relative to all possible 8.6 billion substitutions in the human reference genome. Larger values indicate increased predicted deleteriousness.
    - **raw_score**: Unscaled CADD scores indicating whether a variant is likely to be "observed" (negative values) vs "simulated" (positive values). Larger values indicate increased predicted deleteriousness.
  - **sift**: Sorting Tolerant From Intolerant (SIFT) [score](https://www.nature.com/articles/nprot.2009.86) used to predict the impact of SNVs on protein function.
    - **sift_max**: Scaled probability of the amino acid substitution being tolerated, ranging from 0 to 1. Scores below 0.05 are predicted to impact protein function. We prioritize max scores for MANE Select transcripts where possible and otherwise report a score for the canonical transcript. 
  - **polyphen**: Polymorphism Phenotyping v2 (Polyphen-v2) [score](https://www.nature.com/articles/nmeth0410-248) that predicts the impact of an SNV on protein structure and function.
    - **polyphen_max**: Prediction of the possible impact of an amino acid substitution on the structure and function of a human protein, ranging from 0.0 (tolerated) to 1.0 (deleterious).  We prioritize max scores for MANE Select transcripts where possible and otherwise report a score for the canonical transcript.
  - **phylop**: Per-base evolutionary constraint (score)[https://pubmed.ncbi.nlm.nih.gov/37104612/].
    - **phylop**: Base-wise conservation across the 241 placental mammals in the [Zoonomia](https://zoonomiaproject.org/) project. Score ranges from -20 to 9.28, and reflects acceleration (faster evolution than expected under neutral drift, assigned negative scores) as well as conservation (slower than expected evolution, assigned positive scores). 
  - **splice_ai**: [Score](https://linkinghub.elsevier.com/retrieve/pii/S0092-8674(18)31629-5) that predicts a variant's impact on splicing.
    - **max_ds**: The maximum delta score across 4 splicing consequences, which reflects the probability of the variant being splice-altering. If a variant was predicted to fall within multiple genes, score is across all relevant genes. Score ranges from 0 to 1. In the SpliceAI paper, a detailed characterization of the delta scores is provided for 0.2 (high recall), 0.5 (recommended), and 0.8 (high precision) cutoffs. 
  - **pangolin**: [Score](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-022-02664-4) that predicts a variant's impact on splicing.
    - **largest_ds**: The largest delta score of a variant involved in 2 splicing consequences. If a variant is predicted to fall within multiple genes, score is across all relevant genes. Note that there is different behavior from spliceAI: splice gain is noted as the increase in splice usage (scores range from 0 to 1) and splice loss as the decrease in splice usage (scores range from 0 to -1). The Pangolin authors proposed a cutoff of >|0.2| for the predicted increase or decrease in splice site usage. 
